### DISCUSSION TITLE
Fetch data from another internal table

### DISCUSSION THREAD

**Discussion by sleepsterz** [Budibase/budibase#2200] [Help] [answered]
When reading through the 0.9 changelog I saw that one thing that was stated was "You can also use the formula data type to ‘pluck’ data from another internal table and use that data in a calculation in your present table." Could someone help me out with the formatting required to pull data from another internal table?

**Comment by GuilhermeHirsch** [comment]
Hi, @sleepsterz! How are you? Can you detail the case? I want to replicate it here to show you in detail.

**Reply by sleepsterz** [reply]
Okay for instance you have two tables.

**Table 1**
| Value A | Value B |
| -------- | -------- |
|     0.3    |    0.7     |

**Table 2**
| Value C |
| -------- |
|     0.5    |

If I try to add a calculated field to Table 1 the syntax would be something like {{ add Value A Value B }}.
Now I am trying to figure out what the required syntax would be to pull in the value from Table 2.

**Reply by joebudi** [reply]
Hey - this may help:
https://github.com/helpers/handlebars-helpers#pluck

**Reply by kevmodrome** [reply]
Hi @sleepsterz

You can do this by setting up the two tables and creating a relationship between them. Here's how I set them up:

<img width="862" alt="CleanShot 2021-08-02 at 16 19 33@2x" src="https://user-images.githubusercontent.com/534488/127876321-7dbdcc71-c77a-440e-b76c-afa7cbf49b13.png">

<img width="863" alt="CleanShot 2021-08-02 at 16 19 58@2x" src="https://user-images.githubusercontent.com/534488/127876363-5f72e6b7-8ae6-4c99-b9cf-b806c56cbbac.png">


This was the handlebars expression that I used (Total column, using the Formula type): `{{ add Value A (pluck Relationship to B "Value C") }}`

**Reply by sleepsterz** [reply]
Ah thank you for the assistance. I didn't realize you had to go through a relationship to get to the value on the other table. Unfortunately that wont work for what I wanted to use it for then. Is there any way to set a variable somewhere that can then be referenced in either a calculation or inside the application?

**Reply by kevmodrome** [reply]
Right, so you just want to get at the number in some way in the design section? If I understand correctly one way of solving this would be something like this:

1. Create a table and add a row that contains all the values you want to reach from the design section
2. Create a Data provider and select the table, create a repeater inside of that provider.
3. Add another Data Provider inside the repeater (this would be the "real" table with all your data).
4. Add whatever children you want here.
5. Access the global variable using the repeater you created (it will pop up in the binding drawer) and use it in combination with data form the second data provider that you added

Structure will be something like this:
- Data Provider (dummy table with variables)
     - Repeater (call it "Global" or something)
           - Data Provider (real table with your data)
                  - Children of your choice that can reference the "Global" table as well as the real data

I can try and re-create this for you in a bit if this doesn't help.

**Reply by kevmodrome** [reply]
Oh, and there are some plans to make this a lot easier by creating a sort of "Global" value store that you can reference. Just nothing concretely planned just yet.

**Reply by arminus** [reply]
@kevmodrome is the approach with the nested data providers still valid in 2023? I must admit, I haven't quite grasped the repeater concept yet, my "config table" would be something like a key/value table and I'd need a specific key's value in multiple conditions across my designs.

So that "Global" value store would really come in handy, right now, I have such a config value as literals spread all over the place (which is a maintenance nightmare in case it ever changes)

**Reply by melohagan** [reply, marked as answer]
Hey @arminus If you want to fetch data from another internal table you can do that through the public API: https://docs.budibase.com/reference/post_tables-tableid-rows-search

The OP is quite out of date, I'm not sure the public API was even around back then!
